modules/stochastic_tools/doc/content/source/samplers/CartesianProductSampler.md

```diff
@@ -0,0 +1,29 @@
+# CartesianProductSampler
+
+!syntax description /Samplers/CartesianProductSampler
+
+## Overview
+
+The Cartesian product creates a sample for all possible combinations for a set of values,
+$Z = A \times B \times \cdots \times L \times \ M$. The calculation is being performed in parallel
+using a "lazy" scheme to all for the $n_th$ value to be computed exactly as follows
```

Comment:
I'm not quite sure what it meant by: `using a "lazy" scheme to all for the`.  Maybe clarify that.

Reply:
`$n_th$` -> `$n$th`

Reply:
I don't doubt that this is mathematically correct, but if I didn't already know what this class was about, I would have no clue what you were talking about from this description. I would mention that this is also known as a grid sampler, try to describe in words what it does, and provide an example.